obspy/io/sh/evt.py

```diff
@@ -0,0 +1,282 @@
+        except ValueError as ex:
+            warn(str(ex))
+        else:
+            kws[dest_key] = val
+    return kws
+
+
+def _mags(obj, evid, stamag=False):
+    mags = []
+    pm = None
+    for magtype1, magtype2 in MAG_MAP.items():
+        magkey = 'mean ' * (not stamag) + 'magnitude ' + magtype1
+        if magkey in obj:
+            magv = obj[magkey]
+            if 'inf' in magv:
+                warn('invalid value for magnitude: %s (event id %s)'
+                     % (magv, evid))
+            else:
+                magv = float(magv)
+                mag = (StationMagnitude(station_magnitude_type=magtype2,
+                                        mag=magv) if stamag else
+                       Magnitude(magnitude_type=magtype2, mag=magv))
+                mags.append(mag)
+    if len(mags) == 1:
+        pm = mags[0].resource_id
+    return mags, pm
+
+
+def _seed_id_map(inventory=None, id_map=None, id_default='.{}..{}'):
+    if id_map is None:
+        id_map = {}
+    ret = id_map.copy()
+    if inventory is not None:
+        for net in inventory:
+            for sta in net:
+                if len(sta) == 0:
+                    temp = id_map.get(sta.code, id_default)
+                    temp = temp.split('.', 2)[-1]
+                else:
+                    cha = sta[0]
+                    temp = cha.location_code + '.' + cha.code[:-1] + '{}'
+                ret[sta.code] = net.code + '.{}.' + temp
+    return ret
+
+
+def _read_evt(filename, inventory=None, id_map=None, id_default='.{}..{}'):
+    """
+    Read a SeismicHandler EVT file and returns an ObsPy Catalog object.
+
+    .. warning::
+        This function should NOT be called directly, it registers via the
+        ObsPy :func:`~obspy.core.event.read_events` function, call this
+        instead.
+
+    :type filename: str
+    :param filename: File or file-like object in text mode.
+    :type inventory: :class:`~obspy.core.inventory.inventory.Inventory`
+    :param inventory: Inventory used to retrieve network code, location code
+        and channel code of stations (SEED id).
+    :type id_map: dict
+    :param id_map: If channel information was not found in inventory,
+        it will be looked up in this dictionary
+        (example: `id_map={'MOX': 'GR.{}..HH{}'`).
+        The values must contain three dots and two `{}` which are
+        substituted by station code and component.
+    :type id_default: str
+    :param id_default: Default SEED id expression.
+        The value must contain three dots and two `{}` which are
+        substituted by station code and component.
+
+    :rtype: :class:`~obspy.core.event.Catalog`
+    :return: An ObsPy Catalog object.
+
+    .. note::
+        The following fields are supported by this function: %s.
+
+        Compare with http://www.seismic-handler.org/wiki/ShmDocFileEvt
+    """
+    seed_map = _seed_id_map(inventory, id_map, id_default)
+    with open(filename, 'r') as f:
+        temp = f.read()
```

Comment:
As long as we do Python 2 and 3 it might be safer to read bytes and explicitly decode.. even for pure Python 3 it would be much safer to explicitly specify ASCII or UTF-8 whatever is appropriate? Otherwise we might run into strange issues on some systems (we've seen garbled default system encodings set many times in the past, esp. on Mac ;-P)

Reply:
:+1: